cadc-storage-adapter-swift/src/intTest/java/org/opencadc/inventory/storage/swift/SwiftPreserveNamespaceTest.java

```diff
@@ -85,31 +86,44 @@
 public class SwiftPreserveNamespaceTest extends StorageAdapterBasicTest {
     private static final Logger log = Logger.getLogger(SwiftPreserveNamespaceTest.class);
 
-    static final List<String> PRESERVE = new ArrayList<>();
+    static final List<Namespace> PRESERVE_NAMESPACES = new ArrayList<>();
+    static final List<Namespace> PURGE_NAMESPACES = new ArrayList<>();
     
     static {
         Log4jInit.setLevel("org.opencadc.inventory", Level.INFO);
         Log4jInit.setLevel("org.javaswift.joss.client", Level.INFO);
         
-        PRESERVE.add(StorageAdapterBasicTest.TEST_NAMESPACE);
+        PRESERVE_NAMESPACES.add(new Namespace(StorageAdapterBasicTest.TEST_NAMESPACE));
+        PURGE_NAMESPACES.add(new Namespace(StorageAdapterBasicTest.PURGE_NAMESPACE));
     }
     
     final SwiftStorageAdapter swiftAdapter;
     
     public SwiftPreserveNamespaceTest() throws Exception {
-        super(new SwiftStorageAdapter(true, System.getProperty("user.name") + "-single-test", 2, false, PRESERVE));
+        super(new SwiftStorageAdapter(true, System.getProperty("user.name") + "-single-test", 2, false));
+        adapter.setRecoverableNamespaces(PRESERVE_NAMESPACES);
+        adapter.setPurgeNamespaces(PURGE_NAMESPACES);
         this.swiftAdapter = (SwiftStorageAdapter) super.adapter; 
     }
 
     @Override
     public void cleanupBefore() throws Exception {
         log.info("cleanupBefore: START");
+        
+        // disable recoverable for cleanup
+        List<Namespace> preserveConfig = new ArrayList<>();
+        preserveConfig.addAll(adapter.getRecoverableNamespaces());
```

Comment:
this is to disable recoverable (as set in ctor) during cleanup (line 116) so cleanup really nukes everything, and then restore adapter state on line 126. Otherwise some tests are impossible because of previous state.